File: view/task.py
# -*- coding:utf-8 -*-
#Time:2017-4-27
#
#author:LiMing

from flask import Blueprint,render_template,request,redirect,url_for
import json
from config import db
from module import taskPlan,taskCase
from sqlalchemy import or_,and_

task = Blueprint('task',__name__)

@task.route('/task',methods=['POST','GET'])
def taskList():
    data = db.session.query(taskPlan).all()
    print(data)
    return render_template("task.html",data=data)

@task.route('/getCaseList',methods=['POST','GET'])
def getCaseList():
    if request.method == 'GET':
        data = request.args.get('caselist')
        data=json.loads(data)
        print(data['caselist'])
        return render_template("add_task.html",caselist=data['caselist'])

@task.route('/addTask',methods=['POST','GET'])
def addTask():
    if request.method == 'POST':
        req = request.get_json()
        print(req)
        taskName= req['task_title']
        startTime= req['start_time']
        endTime= req['end_time']
        interval= req['interval']
        desc= req['task_desc']
        task_plan= taskPlan(name=taskName,start_time=startTime,end_time=endTime,interval=interval,desc=desc)
        db.session.add(task_plan)
        db.session.commit()
        task_id = task_plan.id
        for i in req['caseList']:
            caseName=i['title']
            caseId= i['id']
            order= i['order']
            task_case = taskCase(task_id=task_id,case_id=caseId,case_name=caseName,order=order)
            db.session.add(task_case)
            db.session.commit()
        return json.dumps({"status":200,'message':'success!'})

File: static/js/task.js
$(function () {
    $('#case-submit-btn').on('click',function () {
        var data={};
        var task_title = $('#task-name').val();
        var start_time = $('#start-time').val();
        var end_time = $('#end-time').val();
        if(task_title == ""){
            console.log('任务名字不能为空!')
            return
        }
        if(start_time == ""){
            console.log('开始时间不能为空!')
            return
        }
        if(end_time == ""){
            console.log('结束时间不能为空!')
            return
        }
        var interval = $("input[name='optionsRadiosInline']:checked").val();
        var task_desc = $('#task-desc').val();
        var i=0;
        var cList=[];
        $("tr[name='caselist']").each(function () {
            var caseList={};
            var title = $(this).find("td[name='title']").html();
            var project = $(this).find("td[name='project']").html();
            var url = $(this).find("td[name='url']").html();
            var id = $(this).find("td[name='title']").data('id');
            caseList.title = title;
            caseList.project = project;
            caseList.url = url;
            caseList.order=i;
            caseList.id=id;
            i +=1;
            cList.push(caseList);
        });
        data.task_title=task_title;
        data.start_time=start_time;
        data.end_time=end_time;
        data.interval=interval;
        data.task_desc=task_desc;
        data.caseList=cList;
        data=JSON.stringify(data);
        $.ajax({
            contentType: "application/json",
            data: data,
            url: '/addTask',
            type: 'POST',
            dataType: 'json',
            success:function (data) {
                if (data.status == 200){
                    console.log(data.message);
                    self.location="/task";
                }

            }
        });
        console.log(data);
    })
});